plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/PlayAssetDelivery.java

```diff
@@ -205,4 +228,27 @@ public void removePack(String packName, int signalID) {
     removePackTask.addOnSuccessListener(removePackOnSuccessListener);
     removePackTask.addOnFailureListener(removePackOnFailureListener);
   }
+
+  /**
+   * Directly calls showCellularDataConfirmation(Activity activity). The current activity can be
+   * accessed using (Context) getGodot().getApplicationContext(); Shows a confirmation dialog to
+   * resume all pack downloads that are currently in the WAITING_FOR_WIFI state. Emits
+   * showCellularDataConfirmationSuccess and showCellularDataConfirmationError signals when the
+   * underlying task succeeds/fails.
+   *
+   * @param signalID identifier used to track mapping of signals to Tasks
+   */
+  public void showCellularDataConfirmation(int signalID) {
+    OnSuccessListener<Integer> showCellularDataConfirmationSuccessListener =
+        result -> emitSignalWrapper(SHOW_CELLULAR_DATA_CONFIRMATION_SUCCESS, result, signalID);
+    OnFailureListener showCellularDataConfirmationFailureListener =
+        e -> emitSignalWrapper(SHOW_CELLULAR_DATA_CONFIRMATION_ERROR, e.toString(), signalID);
```

Comment:
Rather than convert the exception to a string (`e.toString()`), we should serialize the exception object into a Dictionary and pass that up. Most of these exceptions are going to be [AssetPackExceptions](https://developer.android.com/reference/com/google/android/play/core/assetpacks/AssetPackException) which have a `getErrorCode` method to get the [error code](https://developer.android.com/reference/native/play/core/group/assetpack#assetpackerrorcode) associated with the exception. If we convert the exception to a string, we throw out that information and the GDScript side is going to have to extract that info from the string which is fragile.

We should apply this change to every failure signal.